src/masoniteorm/relationships/HasMany.py

```diff
@@ -28,5 +28,5 @@ def set_keys(self, owner, attribute):
 
     def register_related(self, key, model, collection):
         model.add_relation(
-            {key: collection.where(self.foreign_key, getattr(model, self.local_key))}
+            {key: collection.get(getattr(model, self.local_key)) or Collection()}
```

Comment:
hmm. is this right? before it was fetching by the foreign_key and now by the local_key?

when working with test records and all the records have ids 1-4 it can be difficult to make sure we're fetching the right related records. have to test this